Rakefile

```diff
@@ -1,8 +1,50 @@
-require 'rake/testtask'
+require 'fileutils'
+VERSIONS = %w[ 2.1 2.2 2.3 latest ].freeze
+
+def dcrm_test(version: "latest")
+  puts "\n---\nRunning tests using Docker on Ruby (#{version})"
+  FileUtils.rm_f("Gemfile.lock", verbose: true)
+  FileUtils.rm_f(".bundle/config", verbose: true)
+
+  test = version == "latest" ? "test" \
+    : "test_#{version.gsub(".", "_")}"
+
+  sh("docker-compose run --rm #{test}")
+end
+
+begin
+  require "rake/testtask"
+
+  Rake::TestTask.new do |t|
+    t.pattern = "test/**/*_test.rb"
+  end
+
+rescue LoadError
+  desc "Run tests using Docker on Ruby (latest)"
+  task :test do
+    dcrm_test
+  end
+end
```

Comment:
In the use case for this you don't run bundle locally, but rather inside to the container that's launched, so "rake/testtask" isn't available. That's the determining factor as to wheather a container should be launched or now. 

Because of that, it can't be done at the execution phase.